lib/executor/stage_calculations.go

```diff
@@ -0,0 +1,184 @@
+package executor
+
+import (
+	"math"
+	"time"
+)
+
+// stageTransitionCalculator is a state machine (of sorts) that calculates the transtitions between
+// stages and gives the next full value produced by the stages. In this explanation we are talking
+// about events and in practice those events are either starting of an iteration or starting of a
+// new VU, but could really be anything.
```

Comment:
misleading comment, like we discussed, VU starting isn't ruled by the same logic as iterations